enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts

```diff
@@ -79,23 +82,30 @@ export function useLoadQuestion({
   // Avoid re-running the query if the parameters haven't changed.
   const sqlParameterKey = getParameterDependencyKey(initialSqlParameters);
 
-  const [loadQuestionState, loadQuestion] = useAsyncFn(async () => {
-    const state = await dispatch(
-      runQuestionOnLoadSdk({
+  const [loadQuestionState, loadAndQueryQuestion] = useAsyncFn(async () => {
+    const questionState = await dispatch(
+      loadQuestionSdk({
         options,
         deserializedCard,
         cardId,
-        cancelDeferred: deferred(),
         initialSqlParameters,
       }),
     );
 
-    setQuestionState(state);
+    mergeQuestionState(questionState);
```

Comment:
The problem with the previous version of the code is that we won't know when the query is running i.e. we couldn't rely on `isQueryResultsRunning` since the state will change from not loading question to everything is loaded at once.

This change breaks the chain and update the intermediate state after we've got the first question loaded. This needed to properly render loading and error states.

tl;dr This only changes the intermediate state, but the end states are essentially the same.